ci_framework/roles/dlrn_promote/tasks/main.yml

```diff
@@ -28,8 +28,7 @@
 - name: Run DLRN promote hash
   when:
     - cifmw_dlrn_promote_hash_in_promote_target is defined
-    - "'stdout' in cifmw_dlrn_promote_hash_in_promote_target"
-    - cifmw_dlrn_promote_hash_in_promote_target.stdout == '[]'
+    - cifmw_dlrn_promote_hash_in_promote_target |length == 0 or cifmw_dlrn_promote_hash_in_promote_target == '[]'
```

Comment:
What's the point of the `or` condition if you already compared the length?
PS: the second one will be always false as `cifmw_dlrn_promote_hash_in_promote_target` is a list, not a string like the right side of the operand (but maybe it even fails cause you are comparing different types).

Reply:
Good point, let me remove list string comparison.